Modules/posixmodule.c

```diff
@@ -2712,13 +2712,20 @@ static PyObject *
 os_ttyname_impl(PyObject *module, int fd)
 /*[clinic end generated code: output=c424d2e9d1cd636a input=9ff5a58b08115c55]*/
 {
-    char *ret;
 
-    ret = ttyname(fd);
-    if (ret == NULL) {
+#ifndef TTY_NAME_MAX    /* MacOS doesn't define this macro */
```

Comment:
Huh, maybe it's better to use `sysconf(_SC_TTY_NAME_MAX)` for portability.

Reply:
I wasn't sure about this because I saw the values were quite different between the ones used in OpenBSD and the ones used in Linux, but I agree is the more portable solution